client/src/helpers/navigation.ts

```diff
@@ -0,0 +1,8 @@
+import { Beacon } from "@/types/beacon";
+import { AppRouterInstance } from "next/dist/shared/lib/app-router-context.shared-runtime";
```

Comment:
this import looks fishy. does it still work with a production build? (`pnpm build` and `pnpm start`)